src/mavlink-router/endpoint.cpp

```diff
@@ -172,11 +172,11 @@ int Endpoint::read_msg(struct buffer *pbuf, int *target_sysid, int *target_compi
             return 0;
 
         msg_id = hdr->msgid;
-        payload = rx_buf.data + sizeof(*hdr);
+        pbuf->payload = payload = rx_buf.data + sizeof(*hdr);
         seq = hdr->seq;
-        *src_sysid = hdr->sysid;
-        *src_compid = hdr->compid;
-        payload_len = hdr->payload_len;
+        pbuf->src_sysid = *src_sysid = hdr->sysid;
+        pbuf->src_sysid = *src_compid = hdr->compid;
```

Comment:
src_compid

Reply:
Ugh, I wonder if it became wrong when I rebased or if it was already wrong... It should not have passed the testsuite like this